src/stories/Blocks/header/header-state.js

```diff
@@ -0,0 +1,47 @@
+// Function for checking the size of the desktop links container.
+function desktopLinksContainerCheckSize(container) {
+  // By default, assume we are using the burger menu.
+  document.documentElement.classList.add("has-burger-menu");
+  document.documentElement.classList.remove("has-desktop-menu");
+
+  // The reserved width is for the logo and service buttons.
+  const reservedWidth = 250 + 70 + 70 + 105;
+  // Calculate available space subtracting the reserved width from the total inner window width.
+  const availableSpace = window.innerWidth - reservedWidth;
+
+  // If the width of the menu container is greater than the available space,
+  // we keep the burger menu. Otherwise, we swap to the desktop menu format.
+  if (container.scrollWidth > availableSpace) {
+    document.documentElement.classList.add("has-burger-menu");
+    document.documentElement.classList.remove("has-desktop-menu");
+  } else {
+    document.documentElement.classList.remove("has-burger-menu");
+    document.documentElement.classList.add("has-desktop-menu");
+  }
+}
+
+// Function for initializing the header state based on window size and container width.
+function initHeaderState() {
+  // Finds the node with class 'header__menu-navigation' which is not initialized.
```

Comment:
Cool featue! But shouldn't you wrap it in a document.addEventListener("DOMContentLoaded", () => {} to make sure the markup is ready?

Reply:
Good point, done 👍